UPGRADE-3.x.md

```diff
@@ -1,6 +1,20 @@
 UPGRADE 3.x
 ===========
 
+UPGRADE FROM 3.x to 3.x
+=========================
+
+### Changes in templates registry stucture.
```

Comment:
These messages should explain the user what to do or how resolve possible deprecations.